Python/setup.py

```diff
@@ -39,31 +39,50 @@
 '''
 
 import os
+from pathlib import Path
+import re
 import setuptools
 
+
+def read_requirements():
+    requirements = []
+    with open('./requirements.txt', 'r') as file:
+        for line in file:
+            # exclude comments
+            if re.match('^\\s*#.*', line):
+                # print('****************LINE--', line)
+                continue
+            requirements.append(str(line.lstrip().rstrip()))
```

Comment:
Something more like:

line = line[:line.find ('#')].strip()
if line: requirements.append (line.strip)

More concise and does not require regular expression complexity.

Reply:
That takes off the last char if there's no <CR> but I can do something more concise. ...

Reply:
strip() is supposed to the same as lstrip().rstrip(). Am I missing something?

As written should be append (line) not append (line.strip)

Reply:
Not worried about line continuations. Those two lines throw away everything after and including the '#'

Reply:
You're right. It is the '-1' returned from the find command when the '#' is not in the line that does the damage. The lstrip().rstrip() is my goofiness because I thought strip differed in its removal of \<CR\>s.

Reply:
Ah yes, the -1. Good spot.

Reply:
New code may not be shorter but is more compact lest Pythonic.